### PR TITLE
V3 Split CRUD into Store & Cypher

### PR DESCRIPTION
**Pull request by dgem**
As discussed with @phamann this is _mostly_ a refactor of the current v2 API so that the generation of the cypher queries is separated out and can then be used by the store in order to provide crud for nodes and relationships.

I've also added a couple of test cases and made a few minor tweaks as I found one or two tests to be unreliable.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,31 +2,39 @@
...
+## Testing
+
+Before running the tests you need to build the 'Time Tree'. The tests currently fail second time around, which means before running any tests you should:
+```
+neo4j-shell -c 'match(n) detach delete(n);'
+neo4j-shell -file cypher/time-tree.cql
+neo4j-shell -file cypher/time-tree-index.cql
+```
+
+To make this easier you can just: `cypher/reset-neo4j.sh`
```
Comment (laurieboyes) [inline README.md]: Niiiice

File: test/v2/relationship.spec.js [hunk 2]
```diff
@@ -8,9 +8,11 @@ var request = require('supertest');
 
 var userUuid = require('node-uuid').v4();
 
-describe('API for a graph relationship', function () {
+describe('V3 API for a graph relationship', function () {
```
Comment (laurieboyes) [inline test/v2/relationship.spec.js]: Should these v3 tests be in the v2 directory?

Comment (dgem) [inline test/v2/relationship.spec.js]: hmm, not sure when that happend... I'll check them and fix that !

File: test/v3/caching.spec.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+	{
+		name: 'v2-neo',
+		matcher: process.env.GRAPHENEDB_URL,
+		response:  {
+			body: `[{'uuid': ${userUuid}]`
+		}
+	}
+]);
+
+describe('v2 Caching', function () {
```
Comment (laurieboyes) [inline test/v3/caching.spec.js]: Caching tests in the v3 directory are running tests against the v2 caching stuff. Is this another case of v2/v3 mixup or is this intentional?

Comment (dgem) [inline test/v3/caching.spec.js]: nope, another nice catch... obviously my sed fu was failing me :)

**Comment by laurieboyes** [comment]
LGTM (and passes tests) 👍

**Pull request merged**